WalletWasabi.Gui/Controls/WalletExplorer/HistoryTabViewModel.cs

```diff
@@ -186,35 +176,16 @@ private List<(DateTimeOffset dateTime, Height height, Money amount, string label
 				if (coin.SpenderTransactionId != null)
 				{
 					SmartTransaction foundSpenderTransaction = walletService.TransactionCache.First(x => x.GetHash() == coin.SpenderTransactionId);
-					if (foundSpenderTransaction.Height.Type == HeightType.Chain)
-					{
-						if (walletService.ProcessedBlocks != null) // NullReferenceException appeared here.
-						{
-							if (walletService.ProcessedBlocks.Any(x => x.Value.height == foundSpenderTransaction.Height))
-							{
-								if (walletService.ProcessedBlocks != null) // NullReferenceException appeared here.
-								{
-									dateTime = walletService.ProcessedBlocks.First(x => x.Value.height == foundSpenderTransaction.Height).Value.dateTime;
-								}
-								else
-								{
-									dateTime = DateTimeOffset.UtcNow;
-								}
-							}
-							else
-							{
-								dateTime = DateTimeOffset.UtcNow;
-							}
-						}
-						else
-						{
-							dateTime = DateTimeOffset.UtcNow;
-						}
-					}
-					else
-					{
-						dateTime = foundSpenderTransaction.FirstSeenIfMempoolTime ?? DateTimeOffset.UtcNow;
-					}
+
+					dateTime = foundSpenderTransaction.Height.Type == HeightType.Chain
```

Comment:
Whooow. Let if else be.

Reply:
lol I completely agree that it is easier to read with if else.